CoreFoundation/Base.subproj/CFInternal.h

```diff
@@ -517,53 +519,46 @@ typedef pthread_mutex_t CFLock_t;
 
 #define __CFLockTry(LP) ({ pthread_mutex_trylock(LP) == 0; })
 
-#elif DEPLOYMENT_TARGET_WINDOWS
+#elif _POSIX_THREADS
 
-typedef int32_t CFLock_t;
-#define CFLockInit 0
+typedef pthread_mutex_t CFLock_t;
+#if DEPLOYMENT_TARGET_FREEBSD
+#define CFLockInit ((pthread_mutex_t)PTHREAD_MUTEX_INITIALIZER)
+#else
+#define CFLockInit ((pthread_mutex_t)PTHREAD_ERRORCHECK_MUTEX_INITIALIZER_NP)
+#endif
 #define CF_LOCK_INIT_FOR_STRUCTS(X) (X = CFLockInit)
 
-CF_INLINE void __CFLock(volatile CFLock_t *lock) {
-    while (InterlockedCompareExchange((LONG volatile *)lock, ~0, 0) != 0) {
-	Sleep(0);
-    }
-}
-
-CF_INLINE void __CFUnlock(volatile CFLock_t *lock) {
-    MemoryBarrier();
-    *lock = 0;
-}
+#define __CFLock(LP) ({ (void)pthread_mutex_lock(LP); })
+#define __CFUnlock(LP) ({ (void)pthread_mutex_unlock(LP); })
+#define __CFLockTry(LP) ({ pthread_mutex_trylock(LP) == 0; })
 
-CF_INLINE Boolean __CFLockTry(volatile CFLock_t *lock) {
-    return (InterlockedCompareExchange((LONG volatile *)lock, ~0, 0) == 0);
-}
+typedef int32_t OSSpinLock;
+#define OS_SPINLOCK_INIT 0
+#define OSSpinLockLock(lock) ({ while (__sync_val_compare_and_swap(lock, 0, ~0) != 0) sleep(0); })
+#define OSSpinLockUnlock(lock) ({ __sync_synchronize(); *lock = 0; })
 
-#elif DEPLOYMENT_TARGET_LINUX || DEPLOYMENT_TARGET_FREEBSD
```

Comment:
Why remove this support?

Reply:
Its not really removing support, just merging codepaths to just use the posix threads like on Darwin.  However, Philippe points out that it may be preferable to make them futexes on Linux.